modules/context/auth.go

```diff
@@ -31,10 +31,22 @@ func Toggle(options *ToggleOptions) macaron.Handler {
 		}
 
 		// Check prohibit login users.
-		if ctx.IsSigned && ctx.User.ProhibitLogin {
-			ctx.Data["Title"] = ctx.Tr("auth.prohibit_login")
-			ctx.HTML(200, "user/auth/prohibit_login")
-			return
+		if ctx.IsSigned {
+
+			if ctx.User.ProhibitLogin {
+				ctx.Data["Title"] = ctx.Tr("auth.prohibit_login")
+				ctx.HTML(200, "user/auth/prohibit_login")
+				return
+			}
+
+			if ctx.Req.URL.Path == "/user/change_password" {
```

Comment:
@lafriks Would it just be safe to use `strings.HasSuffix` ? 

Or what is the potential fix here  ? other than giving the url path it's own middleware definition so we can safely handle the scenario without the infinite redirection

Reply:
`strings.HasSuffix(setting.AppSubURL + ctx.Req.URL.Path, "/user/change_password")` ???

Reply:
Just `ctx.Req.URL.Path == setting.AppSubURL + "/user/change_password"`